### PR TITLE
[LDR-R1]EOS-14323:s3recovery fix

### PR DESCRIPTION
**Pull request by shanbhagsv1**

### CODE REVIEW

File: s3recovery/s3recovery/s3recovercorruption.py [hunk 1]
```diff
@@ -116,17 +116,38 @@ def restore_data(self, list_index_id, list_index_id_replica, metadata_index_id,
...
 
         """
-        if ((not self.list_result) or (not self.metadata_result)):
+        self.common_keys = []
+        if ((not self.list_result) and (not self.metadata_result)):
+            self.list_result = {}
+            self.metadata_result = {}
+            return
+
+        if ((not self.list_result) and (not self.metadata_result)):
```
Comment (nileshgovande) [inline s3recovery/s3recovery/s3recovercorruption.py:130]: Duplicate code ?

**Review by nileshgovande — APPROVED** [review]

File: s3recovery/s3recovery/s3recovercorruption.py (hunk 1 above)
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recovercorruption.py]: Done

**Comment by shanbhagsv1** [comment]
Build Job : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/8/

**Review by amitwac2608 — APPROVED** [review]
looks good, gave a minor comment

File: s3recovery/s3recovery/s3recovercorruption.py [hunk 2]
```diff
@@ -116,17 +116,33 @@ def restore_data(self, list_index_id, list_index_id_replica, metadata_index_id,
...
+    def check_consistency(self, list_index_id, list_index_id_replica, metadata_index_id, metadata_index_id_replica):
         """
         Performs consistency check of indexes to be restored
 
         """
-        if ((not self.list_result) or (not self.metadata_result)):
+        self.common_keys = []
+        if ((not self.list_result) and (not self.metadata_result)):
+            self.list_result = {}
+            self.metadata_result = {}
```
[the remark this answers is not in the document]
Comment (shanbhagsv1) [reply s3recovery/s3recovery/s3recovercorruption.py:127]: We are logging "No any data to recover" in this case

**Pull request merged**